src/neo/IO/Helper.cs

```diff
@@ -69,20 +70,18 @@ public static byte[] CompressLz4(this byte[] data)
             int maxLength = LZ4Codec.MaximumOutputSize(data.Length);
             using var buffer = MemoryPool<byte>.Shared.Rent(maxLength);
             int length = LZ4Codec.Encode(data, buffer.Memory.Span);
-            byte[] result = new byte[length];
-            buffer.Memory[..length].CopyTo(result);
+            byte[] result = new byte[length + 4];
+            buffer.Memory[..length].CopyTo(result.AsMemory(4));
+            Array.Copy(BitConverter.GetBytes(data.Length), result, 4);
             return result;
         }
 
         public static byte[] DecompressLz4(this byte[] data, int maxOutput)
         {
-            var maxDecompressDataLength = data.Length * 255;
-            if (maxDecompressDataLength > 0) maxOutput = Math.Min(maxOutput, maxDecompressDataLength);
-            using var buffer = MemoryPool<byte>.Shared.Rent(maxOutput);
-            int length = LZ4Codec.Decode(data, buffer.Memory.Span);
+            int length = BitConverter.ToInt32(data);
             if (length < 0 || length > maxOutput) throw new FormatException();
             byte[] result = new byte[length];
```

Comment:
I guess now we need to check that this `length` still fits into the `PayloadMaxSize` before doing this allocation.

Reply:
Yes, It use maxOutput argument